### PR TITLE
feat(ctb): Remove `setResourceConfig`

### PR DESCRIPTION
**Pull request by clabby**
## Overview

> [!NOTE]
> This PR deliberately does not bump the semantic version. This is done in #10500, these changes are just split up for ease of review.

Removes the `setResourceConfig` setter from the `SystemConfig`, only allowing it to be set by the `ProxyAdmin` owner.

### CODE REVIEW

**Comment by clabby** [comment]
* **#10501** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10501?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#10500** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10500?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clabby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/contracts-bedrock/src/L1/SystemConfig.sol [hunk 1]
```diff
@@ -402,14 +402,6 @@ contract SystemConfig is OwnableUpgradeable, ISemver, IGasToken {
         return _resourceConfig;
     }
 
-    /// @notice An external setter for the resource config.
-    ///         In the future, this method may emit an event that the `op-node` picks up
-    ///         for when the resource config is changed.
```
Comment (mds1) [inline packages/contracts-bedrock/src/L1/SystemConfig.sol:407]: I'm unsure of the origins of this comment, but we may want to move it to the internal `_setResourceConfig` method to preserve it
```solidity
    ///         In the future, this method may emit an event that the `op-node` picks up
    ///         for when the resource config is changed.
```

File: packages/contracts-bedrock/test/L1/SystemConfig.t.sol
Comment (mds1) [inline packages/contracts-bedrock/test/L1/SystemConfig.t.sol:1]: CI failing, looks like we just need `forge fmt`

File: packages/contracts-bedrock/test/L1/SystemConfig.t.sol [hunk 2]
```diff
@@ -209,6 +209,95 @@ contract SystemConfig_Initialize_TestFail is SystemConfig_Initialize_Test {
...
+    ///      is greater than the maximum allowed base fee.
+    function test_setResourceConfig_badMinMax_reverts() external {
+        ResourceMetering.ResourceConfig memory config = ResourceMetering.ResourceConfig({
+            maxResourceLimit: 20_000_000,
+            elasticityMultiplier: 10,
+            baseFeeMaxChangeDenominator: 8,
+            systemTxMaxGas: 1_000_000,
+            minimumBaseFee: 2 gwei,
+            maximumBaseFee: 1 gwei
+        });
```
Comment (maurelian) [inline packages/contracts-bedrock/test/L1/SystemConfig.t.sol:223]: IMO this suite would be easier to read of we passed in the [`DEFAULT_RESOURCE_CONFIG` ](https://github.com/ethereum-optimism/optimism/blob/developer/packages/contracts-bedrock/src/libraries/Constants.sol#L42) values everywhere except for the ones that we're manipulating for testing purposes.

**Review by maurelian — COMMENTED** [review]
Will approve once my and @mds1 comments are addressed.

File: packages/contracts-bedrock/test/L1/SystemConfig.t.sol (hunk 2 above)
Comment (clabby) [reply packages/contracts-bedrock/test/L1/SystemConfig.t.sol:223]: This is the config that the previous tests for the resource config setter were made for. Preferred not modifying them.

Comment (maurelian) [reply packages/contracts-bedrock/test/L1/SystemConfig.t.sol:223]: Disagree, but won't block.

**Review by maurelian — APPROVED** [review]
LGTM once formatted.

**Review by tynes — APPROVED** [review]

**Pull request merged**